README.md

```diff
@@ -1,2 +1,72 @@
 # gnmi-python
-gNMI Python connectivity library. 
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/ambv/black)
+
+This library wraps gNMI functionality to ease usage with Cisco implementations in Python programs. Derived from [openconfig/gnmi](https://github.com/openconfig/gnmi/proto/gnmi/gnmi.proto).
+
+## Usage
+```bash
+# TBD
+#pip install gnmi
+```
+
+This library covers the gNMI defined `capabilities`, `get`, `set`, and `subscribe` RPCs, and helper clients provide OS-specific recommendations. As commonalities and differences are identified this library will be refactored as necessary.
+
+It is *highly* recommended that users of the library learn [Google Protocol Buffers](https://developers.google.com/protocol-buffers/) syntax to significantly ease usage. Understanding how to read Protocol Buffers, and reference the `gnmi.proto`, will be immensely useful for utilizing gNMI and any other gRPC interface.
+
+### Client
+`Client` is a very barebones class simply implementing `capabilities`, `get`, `set`, and `subscribe` methods. It provides some context around the expectation for what should be supplied to these RPC functions and helpers for validation.
+
+Methods are documented in [`gnmi/client.py`](gnmi/client.py).
+
+### XRClient
+`XRClient` inherets from `Client` and provides several wrapper methods which aid with IOS XR-specific behaviors of the gNMI implementation. These are `delete_xpaths`, `get_xpaths`, `set_json`, and `subscribe_xpaths`. These methods make several assumptions about what kind of information will be supplied to them in order to simplify usage of the gNMI RPCs.
+
+Methods are documented in [`gnmi/xr.py`](gnmi/xr.py).
+
+## gNMI
+gRPC Network Management Interface (gNMI) is a service defining an interface for a network management system to interact with a network element. It may be thought of as akin to NETCONF or other control protocols which define operations and behaviors. The scope of gNMI is relatively simple - it seeks to "[[define](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md)] a gRPC-based protocol for the modification and retrieval of configuration from a target device, as well as the control and generation of telemetry streams from a target device to a data collection system. The intention is that a single gRPC service definition can cover both configuration and telemetry - allowing a single implementation on the target, as well as a single NMS element to interact with the device via telemetry and configuration RPCs".
```

Comment:
The word "define" has an extra set of [] around it, did you intend for that?

Reply:
Was indeed the intention. Quoting the spec, wasn't the exact word utilized.